### ISSUE TITLE
Glitched Nintendo Logo in Wave Race 64

### ISSUE THREAD

**Issue by orbea** [GLupeN64/GLupeN64#122]
OS: `Slackware64-current`
`mesa-2016.11.25_af98c6c_master-x86_64-1_git`
`RetroArch-2016.11.27_d6ebd2a_master-x86_64-1_git`
`GLupeN64-2016.11.27_d9fcb20_master-x86_64-1_git`

The nintendo logo when starting Wave Race 64 is graphically glitched and flashes. This occurs with both rsp-hle and rsp-cxd4. It will also occur with both nouveau and the llvmpipe. However it does not really occur in upstream GLideN64 where the logo will blink once with rsp-hle while it works correctly with rsp-cxd4.

When turning off framebuffer emulation neither the issue or the nintendo logo will appear and will show a black screen instead. Additionally with `DEBUG=1` it will also present mostly as a black screen and only flash a little bit.

apitrace - http://ks392457.kimsufi.com/orbea/stuff/trace/libretro/glupen64-waverace.trace.xz
log - http://pastebin.com/QSscgNQR

First bad commit.
```
$ git bisect bad
4378f489285c0bf68e26fd207f47f00d86cab62c is the first bad commit
commit 4378f489285c0bf68e26fd207f47f00d86cab62c
Author: Logan McNaughton <logan@bacoosta.com>
Date:   Wed Oct 12 09:17:32 2016 -0600

    Add LLE RSP Plugin (cxd4)
    
    A good game to test this on is Star Wars: Rogue Squadron

:100644 100644 8e79f0156df6783fc1543dcbc67f60d6b6a0d5bc 06135b9438f81b1584a513280608133a331f2e9f M	Makefile
:100644 100644 89e40a5e38047571289be4fcd9022e82ae0d6356 018f7fd6b083123809e6d1739b337517f604a2c4 M	Makefile.common
:100644 100644 ef54e6735c0b62684b141377db7dbccdf39c609a 6081a1aff7c28f492110d19cac4eb5aaffb192c6 M	README.md
:040000 040000 0fbe7b2b5aeba9ead784b3b287dc2d46a1faadf2 16b623b66e8424e9b1360adb73a0ceb6e69644d7 M	custom
:040000 040000 5a8f97996c745c271724d8ef761f39bed2cd45ef 25d2a387dd67bfce6799df0b20d4331422d47101 M	libretro
:000000 040000 0000000000000000000000000000000000000000 47382713713b7bf7f0cc330ce9038cf4ee1d0a76 A	mupen64plus-rsp-cxd4
```
4378f489285c0bf68e26fd207f47f00d86cab62c
(I confirmed this was the bad commit manually...)

**Comment by diegoviola** [comment]
I can reproduce this bug too.

**Comment by loganmc10** [comment]
Why you say that was the first bad commit what do you mean? That was the first commit with LLE support, so you mean HLE was working before that commit?

**Comment by orbea** [comment]
Yea, before that commit HLE was working as well as upstream. After that commit neither HLE or cxd4 show the nintendo logo correctly in GLupeN64.

**Comment by loganmc10** [comment]
So interestingly enough this seems to be because of the CFLAGS that are passed. It seems like removing ```-mstackrealign``` fixes this issue

Can you confirm by removing that option from these lines?
https://github.com/GLupeN64/GLupeN64/blob/master/Makefile.common#L149
https://github.com/GLupeN64/GLupeN64/blob/master/Makefile.common#L160

I still seem to get 1 blink on both HLE and LLE, is it the same for you? And is that how it is upstream?

**Comment by orbea** [comment]
Yes, that seems to fix it. I removed `-mstackrealign` from the x86_64 section in Makefile.common and it no longer occurs. The single blink occurs upstream with both HLE and LLE for me as well so that seems to be a minor upstream issue. I'm not sure why it wasn't with LLE in upstream last night, but I guess that is not important. I probably just missed it...

**Comment by loganmc10** [comment]
Ok should be fixed by bdd2d0e208ecd2f6221e1e1ed01cab61bba16b9e

**Comment by orbea** [comment]
This is fixed without `DEBUG=1`, while with a debug build it will flash a little and black screen.

apitrace - http://ks392457.kimsufi.com/orbea/stuff/trace/libretro/glupen64_waverace_debug.trace.xz
apitrace log - http://pastebin.com/u9Gy1vgb

Making the trace will show this warning.
`apitrace: warning: glMapBufferRange: MAP_PERSISTENT_BIT w/o FLUSH_EXPLICIT_BIT unsupported (https://github.com/apitrace/apitrace/issues/232)`

As this is an upstream issue as shown in the referenced issue I will close this.
